bgpd/bgp_snmp.c

```diff
@@ -894,6 +894,10 @@ static int bgpTrapEstablished(struct peer *peer)
 	struct in_addr addr;
 	oid index[sizeof(oid) * IN_ADDR_SIZE];
 
+	/* Check if this peer just went to Established */
+	if ((peer->last_major_event != OpenConfirm) || !(peer_established(peer)))
```

Comment:
`peer_established(peer)` should be totally sufficient here, let's simplify to just that.

Reply:
According to the final state machine structure in `bgpd/bgp_fsm.c`, Established->Established is a valid transition (e.g keepalive events), in which case we may not want to call `bgpTrapEstablished`. Hence I check the previous state here, to ensure that I'm replicating the previous behaviour (previously `bgpTrapEstablished` was called when bgp went established from OpenConfirm). Let me know if I'm missing or wrongly assuming something..

Reply:
@qlyoung  _"make the one change I suggested"_  Is the above explanation OK? Let me know if there's anything else I can do.

Reply:
Yes, sorry for the delay.